### PR TITLE
(chore) Set up instructions document how to get started with Docker

### PR DESCRIPTION
**Pull request by tahb**
## Changes in this PR:

* We made a decision to revert the non-docker tooling in order to support the original ADR to use Docker. Even though we currently don't use Docker in production or staging (PaaS now supports it however) we believe there is significant value in using it to help teams get started quickly, and being able to support the live service quickly and confidently.
* Non-docker instructions deleted rather than preserved. Moving them to a docs file would serve no benefit unless someone chose to go against the grain and develop without Docker. If they chose to do this they would very likely find that this file is too out of date and not functional.
* When we come into problems with using Docker in this way for example, slow tests, we want to try to fix the problem before changing the tool. For example we could use docker-sync[1] to improve efficiency when writing to disk which we suspect is the cause of a handful of slow tests.
* Include a brief project summary to start the Readme and introduce the frontend as a closely dependent service

[1] http://docker-sync.io/

### CODE REVIEW

**Comment by jcoglan** [comment]
Thanks for making these changes :) Given that the Docker setup works now, I believe it's true that all one needs to do to boot the app is run `dstart` (or `drebuild` or `dtest-server`). I found a few errors that it would be good to correct:

This phrase only belongs in the frontend docs, not here:

> The API must be running locally before starting this application or you will see a Docker networking error

I'm not sure what file this sentence refers to:

> Copy the contents of a file named `Docker Compose - API - docker-compose.env` into the new docker-compose.env file.

This sentence should probably say "the `ADMIN_EMAILS` variable from `docker-compose.env`:

> It will use the email address from `.env.development / ADMIN_EMAILS`

**Comment by tahb** [comment]
@jcoglan Thanks for reviewing, those bits have been tweaked now. Hopefully the reference to 1password vault connects back to the prerequisite which includes the vault name, otherwise I can repeat the vault name again.

**Comment by tahb** [comment]
@lawrence-dxw Given that this is still open I have added a commit to document how we can safely manage gems.

File: README.md [hunk 1]
```diff
@@ -1,99 +1,78 @@
...
-Once this is done, use Bundler to set up the project:
-
-`$ bundle install`
-
-Create & set up the database:
+Copy the example environment variable file used with Docker:
 
-`$ bundle exec rake db:setup`
+```bash
+$ cp docker-compose.env.sample docker-compose.env
```
Comment (lawrence-dxw) [inline README.md]: What is the purpose of this step - copying the sample file? Would it not be easier to just say that the contents of the file from 1Password should be saved to a file called `<repository-root>/docker-compose.env`?

File: README.md [hunk 2]
```diff
@@ -1,99 +1,78 @@
...
-To run the rspec tests:
+Because the setup and teardown introduces quite some latency, we use the spring service to
+start up all dependencies in a docker container. This makes the test run faster.
+
+Get the test server up and running behind the scenes:
 
-`$ bundle exec rspec`
+```bash
+$ bin/dtest-server up
+$ bin/dtest-server down
```
Comment (lawrence-dxw) [inline README.md]: I think that if I saw this, I would (stupidly, possibly) assume that I was meant to run `up` and then `down`. Should we just put the `up` command here?

**Comment by lawrence-dxw** [comment]
@tahb - very helpful documentation, thanks! Just a couple of minor comments, and a rebase on `develop` needed.

File: README.md (hunk 1 above)
Comment (tahb) [reply README.md]: The intent with the copy command is to provide a small step which gets the application booting successfully, which it should be able to do with nothing more than presence of keys with blank values. 

There is not enough value to hang my hat here on so don't hold this view very strongly!

The value of the approach is perhaps more evident on other projects where 1password was not immediately available and was a blocker to anyone new getting it running whilst their access to whatever secret tool was sorted. Some examples of other repos where we've taken this approach  https://github.com/dfe-digital/teacher-vacancy-service#setting-up-the-project, https://github.com/dxw/mytvh-web#application-setup, https://github.com/dxw/ukti-opportunities#installation, https://github.com/dxw/golden_retriever#installation

File: README.md (hunk 2 above)
Comment (tahb) [reply README.md]: Yes I think that's entirely reasonable and something we should try to avoid!

**Pull request merged**